### ISSUE TITLE
Issues with use as overlay

### ISSUE THREAD

**Issue by mcwitt** [tweag/jupyterWith#103]
There seem to be some issues with the [readme instructions](https://github.com/tweag/jupyterWith#using-as-an-overlay) for using `jupyterWith` as an overlay.

1. The Nix syntax in the example seems incorrect. I think the imports in the overlay list should be of the form
    ``` nix
   import "${path}/nix/haskell-overlay.nix"
    ```
    (as far as I know `++` isn't a valid operator on sets and paths; as it is I get `error: value is a string with context while a list was expected`)

2. I think `path` is meant to refer to the source derivation created by `fetchGit` (not the imported expression), e.g.
    ``` nix
    {
      path = builtins.fetchGit {
        url = https://github.com/tweag/jupyterWith;
        rev = "";
      };
      /* … */
    }
    ```
    (removed surrounding `import`)

3. [overlay.nix](https://github.com/tweag/jupyterWith/blob/b0b4e55da09973a57b82200789816f050a970f3e/nix/overlay.nix#L4) tries to call [default.nix](https://github.com/tweag/jupyterWith/blob/b0b4e55da09973a57b82200789816f050a970f3e/nix/default.nix#L1) with a nonexistent `pkgs` argument.

4. Overlays for haskell and python are already part of [`defaultOverlays`](https://github.com/tweag/jupyterWith/blob/b0b4e55da09973a57b82200789816f050a970f3e/nix/default.nix#L4-L7); it might be misleading to pass them explicitly in the example.

**To Reproduce**
1. With overlay example in `jupyter-with.nix`,
```
$ nix-build jupyter-with.nix
error: value is a set while a list was expected
```

2. Modify import syntax as described above (e.g. `(import "${path}/nix/haskell-overlay.nix")`)
```
$ nix-build jupyter-with-1.nix
error: cannot coerce a set to a string
```

3. Remove `import` around `fetchGit` expression as described above
```
$ nix-build jupyter-with-2.nix
error: anonymous function at /nix/store/0zz1zakm7wqzfz8yixrw04j49chv8km0-source/nix/default.nix:1:1 called with unexpected argument 'pkgs'
```

**Expected behavior**
Example expression should import without errors.

**Environment**
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.4.0, macOS 10.15.4`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.6`
 - channels(matt): `"darwin, home-manager"`
 - channels(root): `"nixpkgs-20.09pre228384.c27e54de99d, darwin, home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by guaraqe** [comment]
You are completely right. It worked someday, but it must have staled after some refactorings. Thanks for reporting it!

I opened a PR (#104) that should fix it. Can you check?

**Comment by mcwitt** [comment]
I just tested 6212dc0 and it fixes the mentioned issues for me. Thanks! I'm finding jupyterWith really useful! :)

**Comment by mcwitt** [comment]
Fixed by #104
